### ISSUE TITLE
Plugins: Background and text color for widgets in line with style in the editor.

### ISSUE THREAD

**Issue by kelsa-pi** [pyzo/pyzo#594]
Hi,
There is no common method to set background and text color for widget in plugins in line with style in the editor.
Best,

PS: 
Using setWidgetStyleSheet(widget) function it is possible to set background and text color for 
particular widget in plugin. 

![Screenshot_2019-03-14_22-49-29](https://user-images.githubusercontent.com/19865053/54394074-20caed80-46ac-11e9-8a78-1dafb776084b.png)

    def setWidgetStyleSheet(self, widget):
        # Style
        theme = pyzo.themes[pyzo.config.settings.theme.lower()]['data']
        background_style = theme['editor.text'].split(',')
        background = background_style[1].split(':')[1]
        textcolor_style = theme['syntax.identifier'].split(',')
        textcolor = textcolor_style[0].split(':')[1]
        wdg = str(type(widget))
        wdg = wdg.replace("<class 'PyQt5.QtWidgets.", "").replace("'>", "")
        ss = wdg + '{background-color:%s; color:%s;}' %  (background, textcolor)
        widget.setStyleSheet(ss)
        widget.setAutoFillBackground(True)

**Comment by almarklein** [comment]
I agree that it would be good to have a way of styling the whole IDE along with the syntax style. One way to do this is:
```py
pyzo.main.setStyleSheet("QWidget {background: #000; color:#fff;}")
```

There are two concerns:
* This may brake Qt trying to conform to the style of the OS in some cases.
* Some elements in Pyzo are currently colored in a way that assumes a light background, and some elements (e.g. the logger shell) even apply their own bg color.

I still think we can and should do this, but it's not trivial.

**Comment by kelsa-pi** [comment]
Hi,
IMHO we have  three option:
1. Plugin developer create his own fuction to set background and text color in line with style in the editor for particular widget in plugin (eg. setBGandTextColor(self, mywidget) 

2. Plugin developer use helper function eg. pyzo.tools.setBGandTextColor(self, mywidget) to set background and text color in line with style in the editor for particular widget if/when needed in plugin. (eg. Source structure tree is not style friendly).

3. IDE  provide a way of styling the whole IDE along with the syntax style eg. pyzo.main.setStyleSheet("QWidget {background: #000; color:#fff;}")

I'm  for option 2.
Option 1 is long shot for styling the IDE, option 3 is hard to implement.
Best

**Comment by almarklein** [comment]
Option 3 might be harder, but would be favorable, I think. It's nice if some tools have a good background, but if the main IDE does not, it still looks bad. I think that Pyzo should be responsible for setting the global style, and plugin developers may need to make sure that things look good on dark as well as light backgrounds.

**Comment by kelsa-pi** [comment]
> Option 3 might be harder, but would be favorable,
+ 1

**Comment by bdieterm** [comment]
Since Qt6, support for dark mode has gotten much better in Pyzo.
And Pyzo itself has also changed much during the last 7 years.

If there is still a problem, please create a new issue.
